backend/src/process.py

```diff
@@ -98,20 +116,41 @@ def enforce_output(raw_output: object, node: NodeData) -> Output:
     return output
 
 
-def run_node(node: NodeData, inputs: Iterable[object], node_id: NodeId) -> Output:
-    assert node.type == "regularNode" or node.type == "iteratorHelper"
+def run_node(
+    node: NodeData, inputs: Iterable[object], node_id: NodeId
```

Comment:
`inputs` can't be an `Iterable`. `Iterable`s do not guarantee that they can be iterated multiple times, which we need to do for error messages.

Reply:
What should I do instead then?

Reply:
nvm, i used List